includes/class-wcsatt-product.php

```diff
@@ -0,0 +1,856 @@
+
+	/**
+	 * Returns a string representing the details of the active subscription scheme.
+	 *
+	 * @param  WC_Product  $product  Product object.
+	 * @param  array       $include  An associative array of flags to indicate how to calculate the price and what to include - @see 'WC_Subscriptions_Product::get_price_string'.
+	 * @param  array       $args     Optional args to pass into 'WC_Subscriptions_Product::get_price_string'. Use 'scheme_key' to optionally define a scheme key to use.
+	 * @return string
+	 */
+	public static function get_price_string( $product, $args = array() ) {
+
+		$scheme_key = isset( $args[ 'scheme_key' ] ) ? $args[ 'scheme_key' ] : '';
+
+		$active_scheme_key = self::get_subscription_scheme( $product );
+		$scheme_key        = '' === $scheme_key ? $active_scheme_key : $scheme_key;
+
+		// Attempt to switch scheme when requesting the price html of a scheme other than the active one.
+		$scheme_switch_required = $scheme_key !== $active_scheme_key;
+		$switched_scheme        = $scheme_switch_required ? self::set_subscription_scheme( $product, $scheme_key ) : false;
+
+		$price_string = WC_Subscriptions_Product::get_price_string( $product, $args );
+
+		// Switch back to the initially active scheme, if switched.
+		if ( $switched_scheme ) {
+			self::set_subscription_scheme( $product, $active_scheme_key );
+		}
+
+		return $price_string;
+	}
+
+	/**
+	 * Returns the price html associated with the active subscription scheme.
+	 * You may optionally pass a scheme key to get the price html string associated with it.
+	 *
+	 * @param  WC_Product  $product     Product object.
+	 * @param  integer     $scheme_key  Scheme key or the currently active one, if undefined. Optional.
+	 * @param  array       $args        Optional args to pass into 'WCS_ATT_Product::get_price_string'. Use 'price_html' to optionally define the bare price html (without subscription details) to use.
+	 * @return string
+	 */
+	public static function get_price_html( $product, $scheme_key = '', $args = array() ) {
+
+		$price_html = isset( $args[ 'price' ] ) ? $args[ 'price' ] : null;
+
+		if ( null === $price_html ) {
+			// No infinite loops, thank you.
+			WCS_ATT_Scheme_Prices::remove_price_html_filters();
+			$price_html = $product->get_price_html();
+			WCS_ATT_Scheme_Prices::add_price_html_filters();
+
+			if ( empty( $price_html ) ) {
+				return $price_html;
+			}
+		}
+
+		$active_scheme_key = self::get_subscription_scheme( $product );
+		$scheme_key        = '' === $scheme_key ? $active_scheme_key : $scheme_key;
+
+		// Attempt to switch scheme when requesting the price html of a scheme other than the active one.
+		$scheme_switch_required = $scheme_key !== $active_scheme_key;
+		$switched_scheme        = $scheme_switch_required ? self::set_subscription_scheme( $product, $scheme_key ) : false;
+
+		// Scheme switch required but unsuccessful? Problem. Set price html to an empty string.
+		if ( $scheme_switch_required && false === $switched_scheme ) {
```

Comment:
Ignore me, I think I was looking at `set_subscription_schemes()` not `set_subscription_scheme()` (i.e. the plural). 🤦‍♂️